### PR TITLE
Create the relationship between class and student

### PR DESCRIPTION
**Pull request by vic778**
All the requirements from phase 1 of the project have been implemented:
## Project requirements

    Create a class Classroom with the following:
        @label instance variable, should be initialized in the constructor.
        Setter and getter for @label (remember about attr_accessor).
    Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
        Create the has-many side (a classroom has many students).
        Create the belongs-to side (a student belongs to a classroom).
        Make sure that when adding a student to a classroom it also sets the classroom for the student.
        Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
    Create a class Book with the following:
        @title and @author instance variables, should be initialized in the constructor.
        Setters and getters for instance variables (remember about attr_accessor).
    Create a class Rental with the following:
        @date instance variable, should be initialized in the constructor.
        Setter and getter for @date (remember about attr_accessor).
    Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
        Create the has-many side of Book and Rental (a book has many rentals).
        Create the belongs-to side of Rental and Book (a rental belongs to a book).
        Create the has-many side of Person and Rental (a person has many rentals).
        Create the belongs-to side of Rental and Person (a rental belongs to a person).
        Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

**Review by BrianSammit — CHANGES_REQUESTED** [review]
# CHANGES REQUIRED 🔴 🛑
Hello @vic778, great job you are doing so far in this project. 💯

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 

However, there are a few changes to make to move to the next project 🛠️ . Keep working hard 💪 💻 

## Required Changes ♻️

- [ ] It would be better if you add the associations `has_many rental` in class person. Kindly fix this minor issue. 

```ruby
class Person
  
  attr_accessor :name, :age, :rentals

 def initialize(age, name = "Unknown", parent_permission = true)
    .
    .
    @rentals = []
  end

  
  def add_rental(book, date)
  
    Rental.new(date, book, self)
  end

end
```

See my inline comment to know what changes are needed. 👇 👀 

Please make these required changes and send it again for another review 👍 💪
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
# HAPPY CODING!!! 😄 

cat


**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

--------

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: student.rb [hunk 1]
```diff
@@ -9,6 +10,11 @@ def initialize(classroom, age, name = 'Unknown', parent_permission: true)
     @parent_permission = parent_permission
   end
 
+  def owner=(classroom)
+    @classroom = classroom
+    classroom.students.push(self) unless classroom.students.include?(self)
+  end
+
```
Comment (mikethreels) [inline student.rb:19]: - [ ] [OPTIONAL] in order to make this work you need to add the attr_reader for `classroom`

File: person.rb [hunk 2]
```diff
@@ -10,8 +14,9 @@ def initialize(age, name = 'Unknown', parrent_permission: true)
     @parrent_permission = parrent_permission
   end
 
-  attr_accessor :name, :age
-  attr_reader :id
+  def add_rental(book, date)
+    Rental.new(book, date, self)
+  end
```
Comment (mikethreels) [inline person.rb:19]: - [ ] as the previous CR already mentioned, in order to make this class work you need to add the attr_acessor for Rental and add it in the `initialize` method as an empty array.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Victoir, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @vic778, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

**Review by Mihndim2020 — COMMENTED** [review]
# STATUS: INVALID!!!  :x: :x: :x: 

Hi @vic778,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code, please make sure your implement the following changes before you resubmit for review. 

## Required Changes ♻️
_Check the comments below._

- [ ] Please you have submitted a project that was already reviewed, approved and merged, kindly create a new repository, redo the project, raise a new `PR` before you submit for review. Please kindly fix that because we can not review an already merged `PR`.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**